lib/context_functions_generator.ex

```diff
@@ -10,6 +10,16 @@ defmodule ContextFunctionsGenerator do
     end
   end
 
+  def generate_function(:exists, name, _pluralized_name, module, opts) do
+    quote do
+      def unquote(:"#{name}_exists?")(id) do
+        Ecto.Query.from(m in unquote(module))
+        |> Ecto.Query.where([m], m.id == ^id)
+        |> unquote(get_repo_module(opts)).exists?()
+      end
+    end
+  end
```

Comment:
Two things here: 

1. We don't need both `where` or `from` here:
```elixir
unquote(module))
|> Ecto.Query.where([x], x.id == ^id)
|> ...
```

OR

```elixir
query = Ecto.Query.from(x in unquote(module), where: x.id == ^id)
unquote(get_repo_module(opts)).exists?(query)
```

2. Using `Ecto.Query` functions like `from` or `where` requires that we `import` or `require` `Ecto.Query` in our modules. Don't know if this is the best behaviour here...
A solution for this is to import the `Ecto.Query` function inside the generated function:

```elixir
iquote do
  def unquote(:"#{name}_exists?")(id) do
    import Ecto.Query, only: [from: 2]
    
    query = Ecto.Query.from(x in unquote(module), where: x.id == ^id)
    
    unquote(get_repo_module(opts)).exists?(query)
  end
end
```

Reply:
If we go with the first option (using piped functions), then I think importing Ecto.Query wouldn't be necessary, right? I think that's the ideal behaviour

Reply:
It's still necessary, since `Ecto.Query.where` is also a macro :/